### ISSUE TITLE
Feature Request: Preserve Original Timestamp

### ISSUE THREAD

**Issue by adamchal** [pornel/ImageOptim#133]
A nice-to-have feature would be: adding the ability to preserve the timestamp of the original image when creating the optimized image.

There are times when both behaviors are desired: the current methodology, where the newly created optimized image has a modified timestamp of now (current behavior), but there are also times when I wish I could preserve the original timestamp in the optimized version.

Thoughts?

**Comment by pornel** [comment]
That's possible. What do you use timestamps for?

**Comment by adamchal** [comment]
One example would be to optimize images (let's say WordPress uploads) that are categorized by directories by date.  It's a little annoying when the file is for date X, but the timestamps are for a much more recent timestamp when doing a `find -mtime` command and you're trying to make sense of things.  I do see how other use cases would want the modified time to be updated after optimization (`rsync` by time only), but the flexibility to enable/disable this feature with a checkbox would be pretty handy.

**Comment by smartbit** [comment]
Would be great, as now I lose all context of the images in a folder. [Stripper](http://www.steelbytes.com/?mid=30) (for Windows) has the option _Preserve file date and times_.

Optionally even leave the EXIF date/time ...

**Comment by pornel** [comment]
ok

**Comment by blorenz** [comment]
This would be nice.

**Comment by infin80** [comment]
+1

**Comment by bkeating** [comment]
Wound up here after realizing I wanted the same feature. +1

**Comment by lauwie007** [comment]
I need this function so I can optimise files from my nas without altering the timestamp (and screwing with the presentation of photos)

Was this ever implemented? Can't find it in version 1.8.8

**Comment by bizmar** [comment]
+1

**Comment by ProBackup-nl** [comment]
@kornelski 

> What do you use timestamps for?

To sort images in Mac OS X finder. My camera import module imports images with Finder column "Date added" set to the date and time added on to the computer file system, and "Date Modified" is set to "date and time as shot".

Especially keeping "Date Modified" is important.

**Comment by ylluminate** [comment]
Yes, there are several folks I work with that all have the same complaint about losing dates and not using ImageOptim for this very reason. It would be so nice to retain especially `mtime`. *All* attributes really should remain untouched if so desired.

**Comment by codename-osni** [comment]
> I need this function so I can optimise files from my nas without altering the timestamp (and screwing with the presentation of photos)
> 
> Was this ever implemented? Can't find it in version 1.8.8

Same use case and question here.
It's now been 5 years now since this feature was requested, but I'm still hopeful!

**Comment by pranaygajawada** [comment]
+1 Have been waiting for this for a long time. Can you please provide this feature?

**Comment by bkeating** [comment]
This can be closed, Look above at the merged PR by @lgarron (thank you!)

EDIT: not seeing it in the `1.8.8` release that is available via the website

**Comment by ylluminate** [comment]
I would assume that the most recent 1.9.3 that was installed by the updater has this patch in it. 

@bkeating just download the beta build (https://imageoptim.com/ImageOptim1.9.2-b2.zip) and then use the built in updater to get the most recent release.

**Comment by nitrodox** [comment]
Unfortunately, right now the checkbox title can't be translated, I suppose. I couldn't find any string in resources on Transifex.
![Снимок экрана 2024-01-15 в 21 04 23](https://github.com/ImageOptim/ImageOptim/assets/153652995/2995c882-413d-4c98-9770-18ed5b8214fb)
